keps/sig-node/3675-resource-plugin-manager/README.md

```diff
@@ -0,0 +1,1002 @@
+
+#### Operational Requirements for Alpha:
+
+* Enable CCI Feature Gate
+* Disable Memory manager 
+* Disable Topology Manager
+* Set CPU Manager policy to None
+
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: CCIResourceManager
+  - Components depending on the feature gate: CCIResourceManager, Kubelet
+- [ ] Other
+  - Describe the mechanism:
+  - Will enabling / disabling the feature require downtime of the control
+    plane?
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+
+###### Does enabling the feature change any default behavior?
+
+No
+
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes, a deletion of pods requiring cci driver will be recommended
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Running pods should not be impacted.  New pods should be handled by the CCI Driver.
+
+###### Are there any tests for feature enablement/disablement?
+
+Yes.
+
+### Rollout, Upgrade and Rollback Planning
+
+The usual Kubernetes upgrade and downgrade strategy applies for in-tree components. 
+Vendors must take care that upgrades and downgrades work with the drivers that
+they provide to customers
+  
+Pods which require CCI Driver will fail starting and report an error due to driver 
+unavailability.  If a CCI Driver fails, Pods which were about to be allocated
+that were tied to the driver will fail. Based on requirements a fallback mechanism 
+can be implemented where such Pods fallback to a given std. QoS Model.
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+Failure of CCI drivers will have impact only over pods requiring CCI driver for 
+resource management. All other pods will not be impacted. THe impacted pods will
+an error message showing failure of the driver.
+
+
+###### What specific metrics should inform a rollback?
+
+Unhealthy CCI drivers which can't get to a ready state and repeated failures 
+to allocate pods in cases of free resources.
+
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+TBD in Beta
+
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+TBD in Beta
+
+
+### Monitoring Requirements
+
+TBD in Beta
+
+
+###### How can an operator determine if the feature is in use by workloads?
+
+Pods will be running on the cluster which are associated to CCI drivers. 
+Additionally, available resources will be updated.
+
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [ ] Events
+  - Event Reason: 
+- [x] API .status
+  - Condition name: CCI Driver readiness and CCI Status field can be introduced
```

Comment:
This field wasn't described in the design section. Can you please elaborate about CCIDriver readiness and the status field. By CCIDriver readiness do you mean the running CCIDriver pod? How is the status populated, updated and the components responsible for populating/updating it.

Reply:
correct, this was meant as the cci driver pod. populated by kubelet